{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": [],
      "authorship_tag": "ABX9TyOhV/qAU24QU7fk7NyN5RUe",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/stockman2/BIOINF_influenza/blob/main/HW2.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "freeman@mashine:~/Workplace/Prac$ samtools flagstat roommate.bam\n",
        "361349 + 0 in total (QC-passed reads + QC-failed reads)\n",
        "358265 + 0 primary\n",
        "0 + 0 secondary\n",
        "3084 + 0 supplementary\n",
        "0 + 0 duplicates\n",
        "0 + 0 primary duplicates\n",
        "361116 + 0 mapped (99.94% : N/A)\n",
        "358032 + 0 primary mapped (99.93% : N/A)\n",
        "0 + 0 paired in sequencing\n",
        "0 + 0 read1\n",
        "0 + 0 read2\n",
        "0 + 0 properly paired (N/A : N/A)\n",
        "0 + 0 with itself and mate mapped\n",
        "0 + 0 singletons (N/A : N/A)\n",
        "0 + 0 with mate mapped to a different chr\n",
        "0 + 0 with mate mapped to a different chr (mapQ>=5)"
      ],
      "metadata": {
        "id": "m6yiVmNdWMHc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp my.mpileup \n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from my.mpileup\n",
        "Chrom\tPosition\tRef\tVar\tCons:Cov:Reads1:Reads2:Freq:P-value\tStrandFilter:R1+:R1-:R2+:R2-:pval\tSamplesRef\tSamplesHet\tSamplesHom\tSamplesNC\tCons:Cov:Reads1:Reads2:Freq:P-value\n",
        "KF848938.1\t72\tA\tG\tG:16794:6:16787:99.96%:0E0\tPass:4:2:10898:5889:1E0\t0\t0\t1\t0\tG:16794:6:16787:99.96%:0E0\n",
        "KF848938.1\t117\tC\tT\tT:20663:36:20625:99.82%:0E0\tPass:27:9:13462:7163:1E0\t0\t0\t1\t0\tT:20663:36:20625:99.82%:0E0\n",
        "KF848938.1\t774\tT\tC\tC:38191:8:38177:99.96%:0E0\tPass:7:1:19806:18371:1E0\t0\t0\t1\t0\tC:38191:8:38177:99.96%:0E0\n",
        "KF848938.1\t999\tC\tT\tT:32115:40:32070:99.86%:0E0\tPass:23:17:17184:14886:1E0\t0\t0\t1\t0\tT:32115:40:32070:99.86%:0E0\n",
        "KF848938.1\t1260\tA\tC\tC:23033:2:23019:99.94%:0E0\tPass:0:2:9824:13195:1E0\t0\t0\t1\t0\tC:23033:2:23019:99.94%:0E0\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp my.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from my.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ samtools mpileup -f reference.fasta  roommate.bam -d 0 >  roommate.mpileup\n",
        "[mpileup] 1 samples in 1 input files\n",
        "[mpileup] Max depth set to maximum value (2147483647)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "\n",
        "\n",
        "freeman@mashine:~/Workplace/Pracjava -jar /home/freeman/Downloads/VarScan.jar mpileup2snp --min-var-freq 0.95 roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.95\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Input stream not ready, waiting for 5 seconds...\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "^Cfreeman@mashine:~/Workplace/Pracsamtools mpileup -f reference.fasta  roommate.bam -d 0 >  roommate.mpileup\n",
        "[mpileup] 1 samples in 1 input files\n",
        "[mpileup] Max depth set to maximum value (2147483647)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.01 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.01\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "6 variant positions (5 SNP, 1 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0001 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t1.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "1060 variant positions (1039 SNP, 21 indel)\n",
        "6 were failed by the strand-filter\n",
        "1033 variant positions reported (1033 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0005 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t5.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "23 variant positions (21 SNP, 2 indel)\n",
        "0 were failed by the strand-filter\n",
        "21 variant positions reported (21 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0003 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t3.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "201 variant positions (194 SNP, 7 indel)\n",
        "2 were failed by the strand-filter\n",
        "192 variant positions reported (192 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.001 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.001\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "23 variant positions (21 SNP, 2 indel)\n",
        "0 were failed by the strand-filter\n",
        "21 variant positions reported (21 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.01 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.01\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "6 variant positions (5 SNP, 1 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ "
      ],
      "metadata": {
        "id": "CPYlHiHqUi_6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp my.mpileup \n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from my.mpileup\n",
        "Chrom\tPosition\tRef\tVar\tCons:Cov:Reads1:Reads2:Freq:P-value\tStrandFilter:R1+:R1-:R2+:R2-:pval\tSamplesRef\tSamplesHet\tSamplesHom\tSamplesNC\tCons:Cov:Reads1:Reads2:Freq:P-value\n",
        "KF848938.1\t72\tA\tG\tG:16794:6:16787:99.96%:0E0\tPass:4:2:10898:5889:1E0\t0\t0\t1\t0\tG:16794:6:16787:99.96%:0E0\n",
        "KF848938.1\t117\tC\tT\tT:20663:36:20625:99.82%:0E0\tPass:27:9:13462:7163:1E0\t0\t0\t1\t0\tT:20663:36:20625:99.82%:0E0\n",
        "KF848938.1\t774\tT\tC\tC:38191:8:38177:99.96%:0E0\tPass:7:1:19806:18371:1E0\t0\t0\t1\t0\tC:38191:8:38177:99.96%:0E0\n",
        "KF848938.1\t999\tC\tT\tT:32115:40:32070:99.86%:0E0\tPass:23:17:17184:14886:1E0\t0\t0\t1\t0\tT:32115:40:32070:99.86%:0E0\n",
        "KF848938.1\t1260\tA\tC\tC:23033:2:23019:99.94%:0E0\tPass:0:2:9824:13195:1E0\t0\t0\t1\t0\tC:23033:2:23019:99.94%:0E0\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp my.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from my.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ samtools mpileup -f reference.fasta  roommate.bam -d 0 >  roommate.mpileup\n",
        "[mpileup] 1 samples in 1 input files\n",
        "[mpileup] Max depth set to maximum value (2147483647)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar --min-var-freq 0.95 mpileup2snp roommate.mpileup > VarScan_results.vcf\n",
        "Command not recognized\n",
        "VarScan v2.4\n",
        "\n",
        "***NON-COMMERCIAL VERSION***\n",
        "\n",
        "USAGE: java -jar VarScan.jar [COMMAND] [OPTIONS] \n",
        "\n",
        "COMMANDS:\n",
        "\tpileup2snp\t\tIdentify SNPs from a pileup file\n",
        "\tpileup2indel\t\tIdentify indels a pileup file\n",
        "\tpileup2cns\t\tCall consensus and variants from a pileup file\n",
        "\tmpileup2snp\t\tIdentify SNPs from an mpileup file\n",
        "\tmpileup2indel\t\tIdentify indels an mpileup file\n",
        "\tmpileup2cns\t\tCall consensus and variants from an mpileup file\n",
        "\n",
        "\tsomatic\t\t\tCall germline/somatic variants from tumor-normal pileups\n",
        "\tcopynumber\t\t\tDetermine relative tumor copy number from tumor-normal pileups\n",
        "\treadcounts\t\tObtain read counts for a list of variants from a pileup file\n",
        "\n",
        "\tfilter\t\t\tFilter SNPs by coverage, frequency, p-value, etc.\n",
        "\tsomaticFilter\t\tFilter somatic variants for clusters/indels\n",
        "\tfpfilter\t\tApply the false-positive filter\n",
        "\n",
        "\tprocessSomatic\t\tIsolate Germline/LOH/Somatic calls from output\n",
        "\tcopyCaller\t\tGC-adjust and process copy number changes from VarScan copynumber output\n",
        "\tcompare\t\t\tCompare two lists of positions/variants\n",
        "\tlimit\t\t\tRestrict pileup/snps/indels to ROI positions\n",
        "\n",
        "\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp --min-var-freq 0.95 roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.95\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "^Cfreeman@mashine:~/Workplace/Pracjava -jar /home/freeman/Downloads/VarScan.jar mpileup2snp --min-var-freq 0.95 roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.95\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Input stream not ready, waiting for 5 seconds...\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "^Cfreeman@mashine:~/Workplace/Pracsamtools mpileup -f reference.fasta  roommate.bam -d 0 >  roommate.mpileup\n",
        "[mpileup] 1 samples in 1 input files\n",
        "[mpileup] Max depth set to maximum value (2147483647)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar /home/freeman/Downloads/VarScan.jar mpileup2snp roommate.mpileup > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "^Cfreeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.2\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.01 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.01\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "6 variant positions (5 SNP, 1 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0001 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t1.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "1060 variant positions (1039 SNP, 21 indel)\n",
        "6 were failed by the strand-filter\n",
        "1033 variant positions reported (1033 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0005 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t5.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "23 variant positions (21 SNP, 2 indel)\n",
        "0 were failed by the strand-filter\n",
        "21 variant positions reported (21 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.0003 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t3.0E-4\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "201 variant positions (194 SNP, 7 indel)\n",
        "2 were failed by the strand-filter\n",
        "192 variant positions reported (192 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.001 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.001\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "23 variant positions (21 SNP, 2 indel)\n",
        "0 were failed by the strand-filter\n",
        "21 variant positions reported (21 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.01 > VarScan_results_roommate.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.01\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "6 variant positions (5 SNP, 1 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ \n"
      ],
      "metadata": {
        "id": "0tFNVEWahQBw"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.01 > VarScan_results_roommate.html\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.01\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "6 variant positions (5 SNP, 1 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.99 > VarScan_results_roommate.html\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.99\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ java -jar VarScan.jar mpileup2snp roommate.mpileup --min-var-freq 0.99 > VarScan_results_roommate_2.vcf\n",
        "Only SNPs will be reported\n",
        "Warning: No p-value threshold provided, so p-values will not be calculated\n",
        "Min coverage:\t8\n",
        "Min reads2:\t2\n",
        "Min var freq:\t0.99\n",
        "Min avg qual:\t15\n",
        "P-value thresh:\t0.01\n",
        "Reading input from roommate.mpileup\n",
        "1665 bases in pileup file\n",
        "5 variant positions (5 SNP, 0 indel)\n",
        "0 were failed by the strand-filter\n",
        "5 variant positions reported (5 SNP, 0 indel)\n",
        "freeman@mashine:~/Workplace/Prac$ cat VarScan_results_roommate_2.vcf | awk 'NR>24 {print $1, $2, $4, $5}'\n",
        "freeman@mashine:~/Workplace/Prac$ cat VarScan_results_roommate_2.vcf | awk 'NR>1 {print $1, $2, $4, $5}'\n",
        "KF848938.1 72 G G:16794:6:16787:99.96%:0E0\n",
        "KF848938.1 117 T T:20663:36:20625:99.82%:0E0\n",
        "KF848938.1 774 C C:38191:8:38177:99.96%:0E0\n",
        "KF848938.1 999 T T:32115:40:32070:99.86%:0E0\n",
        "KF848938.1 1260 C C:23033:2:23019:99.94%:0E0\n",
        "freeman@mashine:~/Workplace/Prac$ cat VarScan_results_roommate_2.vcf | awk '{print $1, $2, $4, $5}'\n",
        "Chrom Position Var Cons:Cov:Reads1:Reads2:Freq:P-value\n",
        "KF848938.1 72 G G:16794:6:16787:99.96%:0E0\n",
        "KF848938.1 117 T T:20663:36:20625:99.82%:0E0\n",
        "KF848938.1 774 C C:38191:8:38177:99.96%:0E0\n",
        "KF848938.1 999 T T:32115:40:32070:99.86%:0E0\n",
        "KF848938.1 1260 C C:23033:2:23019:99.94%:0E0\n",
        "freeman@mashine:~/Workplace/Prac$ cat VarScan_results_roommate_2.vcf | awk 'NR>1 {print $1, $2, $4, $5}'\n",
        "KF848938.1 72 G G:16794:6:16787:99.96%:0E0\n",
        "KF848938.1 117 T T:20663:36:20625:99.82%:0E0\n",
        "KF848938.1 774 C C:38191:8:38177:99.96%:0E0\n",
        "KF848938.1 999 T T:32115:40:32070:99.86%:0E0\n",
        "KF848938.1 1260 C C:23033:2:23019:99.94%:0E0\n"
      ],
      "metadata": {
        "id": "pYxZLKxyhQ9H"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "awk '{ {count++; bases += $4} } END{print bases/count}' roommate.mpileup\n",
        "awk '{if(NR%4==2) {bases += length} } END{print bases/1665}' roommate.fastq # где 1665 - размер reference последовательности"
      ],
      "metadata": {
        "id": "NxUReCncaExS",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 130
        },
        "outputId": "7faf1fcd-822e-4864-bdaa-bdf8f55040ed"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "error",
          "ename": "SyntaxError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-1-c8b10411995d>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    awk '{ {count++; bases += $4} } END{print bases/count}' roommate.mpileup\u001b[0m\n\u001b[0m                                                          ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
          ]
        }
      ]
    }
  ]
}